modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4724,12 +4724,15 @@ public CodegenOperation fromOperation(String path,
 
         CodegenParameter bodyParam = null;
         RequestBody requestBody = operation.getRequestBody();
+
+        Boolean handleFormContent = additionalProperties.get(CodegenConstants.HANDLE_FORM_CONTENT) != null ? (Boolean) additionalProperties.get(CodegenConstants.HANDLE_FORM_CONTENT) : false;
+
         if (requestBody != null) {
             String contentType = getContentType(requestBody);
             if (contentType != null) {
                 contentType = contentType.toLowerCase(Locale.ROOT);
             }
-            if (contentType != null &&
+            if (!handleFormContent && contentType != null &&
```

Comment:
can you please elaborate more on this change?

Reply:
this is a new flag to disable the logic on how the form params are being handled, currently the body is being consumed and all identified form params extracted, in this branch `bodyParam` is not being set which is required by the axum template:

openapi-generator/modules/openapi-generator/src/main/resources/rust-axum/server-operation.mustache

```mustache
  {{#bodyParam}}
    {{#vendorExtensions}}
      {{#x-consumes-json}}
          Json(body): Json<{{^required}}Option<{{/required}}{{{dataType}}}{{^required}}>{{/required}}>,
      {{/x-consumes-json}}
      {{#x-consumes-form-urlencoded}}
          Form(body): Form<{{^required}}Option<{{/required}}{{{dataType}}}{{^required}}>{{/required}}>,
      {{/x-consumes-form-urlencoded}}
```

axum can deserialize form params from the body

https://docs.rs/axum/latest/axum/struct.Form.html

Reply:
I prefer using `!(this instanceof RustAxumServerCodegen)` to make it much easier to understand this logic is tailor-made for the RustAxumServerCodegen.

Reply:
it does work, however this to me looks that we create a dependency to more specialized class inside the generic one (since we have to import `RustAxumServerCodegen`)

but I see that similar was done before.

```java
import org.openapitools.codegen.languages.PhpNextgenClientCodegen;
import org.openapitools.codegen.languages.RustAxumServerCodegen; // new import
import org.openapitools.codegen.languages.RustServerCodegen;
```